src/lib/npm.js

```diff
@@ -25,22 +22,65 @@ async function getNpmToken({npm, options}) {
   };
 
   const uri = url.resolve(npm.registry, '-/user/org.couchdb.user:' + encodeURIComponent(npm.username));
-  const {err, token} = await client.requestAsync(uri, {method: 'PUT', body}).catch(err => {
-    // Some registries (Sinopia) return 409 for existing users, retry using authenticated call
-    if (err.code === 'E409') {
-      return client
-        .requestAsync(uri, {authed: true, method: 'PUT', auth: {username: npm.username, password: npm.password}, body})
-        .catch(err => ({err}));
-    }
+
+  const {err, token} = await new Promise(resolve => {
```

Comment:
Why changing to a callback call here?
You could use `async/await` instead:
```js
let err;
let token;
try {
({err, token} = await client.requestAsync(uri, {method: 'PUT', body}));
} catch (err) {
if (err.code === 'E409') {
....
} else if (....) {
....
}

```

Reply:
The reason for using a callback is that Promises (either directly or through promisify) don't support getting multiple arguments back on failure. Only `err` is accessible. I need response.headers too. So the only way is to use the callback, and wrap it myself.

Reply:
Ah good point. For info you can solve that by using [pify with the multiargs](https://github.com/sindresorhus/pify#multiargs) option.
But that would be something for another PR

Reply:
I searched for a lot of libraries, including bluebird and es6-promisify, but they both support multiArgs only on success, not on error. Thank you for pointing me at this, I will create an issue for it.